components/de_amp/browser/test/de_amp_service_unittest.cc

```diff
@@ -0,0 +1,138 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/de_amp/browser/de_amp_service.h"
+#include "components/prefs/testing_pref_service.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "third_party/googletest/src/googletest/include/gtest/gtest.h"
+
+namespace de_amp {
+
+class DeAmpServiceUnitTest : public testing::Test {
+ public:
+  DeAmpServiceUnitTest() {}
+  ~DeAmpServiceUnitTest() override = default;
+
+  void SetUp() override {
+    pref_service_ = std::make_unique<TestingPrefServiceSimple>();
+    DeAmpService::RegisterProfilePrefs(pref_service_->registry());
+    service_ = std::make_unique<DeAmpService>(pref_service_.get());
+  }
+
+  bool CheckIfAmpDetected(const std::string body, std::string* canonical_link) {
+    return service_->FindCanonicalLinkIfAMP(body, canonical_link);
```

Comment:
In fact we don't need `service_` and `pref_service_` to check this, it's a static member.
Better to make such tests without a fixture (`TEST_F` => `TEST`)